src/components/MovieForm/MovieForm.js

```diff
@@ -0,0 +1,310 @@
+    }
+];
+
+class MovieForm extends React.Component {
+    state = {
+        form: this.props.movieData ? {
+            movieTitle: this.props.movieData.movieTitle || '',
+            releaseDate: this.props.movieData.releaseDate || '2024-03-08',
+            movieUrl: this.props.movieData.movieUrl || '',
+            rating: this.props.movieData.rating || 0,
+            relevantGenres: this.props.movieData.relevantGenres.map((genre) => {
+                return GENRE_LIST.find((item) => item.value === genre.value);
+            }).filter(Boolean)|| [],
+            duration: this.props.movieData.duration || 0,
+            description: this.props.movieData.description || '',
+        } : EMPTY_FORM,
+        genreList: [
+            {   value: 'All',
+                label: 'All',
+            },
+            {
+                value: 'Documentary',
+                label: 'Documentary'
+            },
+            {
+                value: 'Comedy',
+                label: 'Comedy'
+            },
+            {
+                value: 'Thriller',
+                label: 'Thriller'
+            },
+            {
+                value: 'Crime',
+                label: 'Crime',
+            },
+            {
+                value: 'Action',
+                label: 'Action',
+            },
+            {
+                value: 'Drama',
+                label: 'Drama',
+            }
+        ]
+    };
+
+    resetForm = (e) => {
+        e.preventDefault()
+        this.setState({
+            form: EMPTY_FORM,
+        });
+    }
+
+    onTitleChange = (e) => {
+        this.setState({
+            form: {
+                ...this.state.form,
+                movieTitle: e.target.value,
+            }
+        });
+    }
+    onReleaseDateChange = (e) => {
+        this.setState({
+            form: {
+                ...this.state.form,
+                releaseDate: e.target.value,
+            }
+        });
+    }
+    onMovieUrlChange = (e) => {
+        this.setState({
+            form: {
+                ...this.state.form,
+                movieUrl: e.target.value,
+            }
+        });
+    }
+    onRatingChange = (e) => {
+        this.setState({
+            form: {
+                ...this.state.form,
+                rating: e.target.value,
+            }
+        });
+    }
+    onGenreChange = (e) => {
+        this.setState({
+            form: {
+                ...this.state.form,
+                relevantGenres: e,
+            },
+        })
+    }
+    onDurationChange = (e) => {
+        this.setState({
+            form: {
+                ...this.state.form,
+                duration: e.target.value,
+            }
+        });
+    }
+    onDescriptionChange = (e) => {
+        this.setState({
+            form: {
+                ...this.state.form,
+                description: e.target.value,
+            }
+        });
+    }
+
+    render() {
+        return (
+            <form
```

Comment:
You can use uncontrolled form if you don't have a complex validation.

![image](https://github.com/mradomski-epam/reactjs-mentoring/assets/118353174/c10ac7cb-b18b-4fa0-80c3-caae8e36f4e2)

If you still want/need to control a form, you can use your input's name in on change handler, so you will have only 1 handler for controlling the state.

Example:
```js
  onChange = (e) => {
        const {name, value} = e.target;
        this.setState({
            form: {
                ...this.state.form,
                [name]: value
            }
        })
    }
```
(...)
```js
          <input
            className="MovieForm__input"
            name="movieTitle"
            id="movieTitle"
            type="text"
            placeholder="Movie title"
            onChange={this.onInputChange}
            value={this.state.form.movieTitle}
          />
          <input
            className="MovieForm__input MovieForm__input-number"
            name="releaseDate"
            id="releaseDate"
            type="date"
            placeholder="Select date"
            onChange={this.onInputChange}
            value={this.state.form.releaseDate}
          />
```